lib/js/components/Drawer/DrawerSection/DrawerSection.vue

```diff
@@ -1,106 +1,137 @@
-			}
-		}
-
-		&.-isExpandable {
-			cursor: pointer;
-		}
-	}
-
-	&__headerTitleWrapper {
-		align-items: center;
-		display: flex;
-		gap: $space-xxs;
-		justify-content: space-between;
-	}
-
-	&__headerTitle {
-		@include info-s-extensive-bold-uppercase;
-
-		color: $color-neutral-text-weak;
-		padding: $space-xxs 0;
-	}
-}
-</style>
-
 <script lang="ts">
-import Divider from '../../Divider';
-import IconButton from '../../Buttons/IconButton/IconButton.vue';
-import { ICONS } from '../../Icons/Icon';
+import { IconItem, ICONS } from '../../Icons/Icon';
 import { ICON_BUTTON_COLORS, ICON_BUTTON_SIZES } from '../../Buttons/IconButton';
+import SectionHeader, {
+	SECTION_HEADER_ICON_COLORS,
+	SECTION_HEADER_MOBILE_LAYOUTS,
+	SECTION_HEADER_SIZES,
+	SectionHeaderIconColor,
+} from '../../Headers/SectionHeader';
+import { toRaw } from 'vue';
 
 export default {
 	name: 'DrawerSection',
 	components: {
-		DsIconButton: IconButton,
-		DsDivider: Divider,
+		DsSectionHeader: SectionHeader,
 	},
 	props: {
-		hasDivider: {
+		isExpandable: {
+			type: Boolean,
+			default: false,
+		},
+		hideSlotWhenCollapsed: {
+			type: Boolean,
+			default: false,
+		},
+		iconLeft: {
+			type: Object as () => IconItem,
+			default: null,
+			validator(iconLeft: IconItem) {
+				return Object.values(ICONS).includes(toRaw(iconLeft));
+			},
+		},
+		iconLeftColor: {
+			type: String as () => SectionHeaderIconColor,
+			default: SECTION_HEADER_ICON_COLORS.NEUTRAL_WEAK,
+			validator(iconLeftColor: SectionHeaderIconColor) {
+				return Object.values(SECTION_HEADER_ICON_COLORS).includes(toRaw(iconLeftColor));
+			},
+		},
+		iconRight: {
+			type: Object as () => IconItem,
+			default: null,
+			validator(iconRight: IconItem) {
+				return Object.values(ICONS).includes(toRaw(iconRight));
+			},
+		},
+		iconRightColor: {
+			type: String as () => SectionHeaderIconColor,
+			default: SECTION_HEADER_ICON_COLORS.NEUTRAL_WEAK,
+			validator(iconRightColor: SectionHeaderIconColor) {
+				return Object.values(SECTION_HEADER_ICON_COLORS).includes(toRaw(iconRightColor));
+			},
+		},
+		isExpanded: {
 			type: Boolean,
 			default: true,
 		},
-		isExpandable: {
+		info: {
 			type: Boolean,
 			default: false,
 		},
+		size: {
+			type: String,
+			default: SECTION_HEADER_SIZES.XX_SMALL,
+			validator(size) {
+				return Object.values(SECTION_HEADER_SIZES).includes(size);
+			},
+		},
 		title: {
+			type: String,
+			required: true,
+		},
+		eyebrow: {
+			type: String,
+			default: '',
+		},
+		supportingText: {
 			type: String,
 			default: null,
 		},
+		divider: {
+			type: Boolean,
+			default: true,
+		},
+		mobileLayout: {
+			type: String,
+			default: SECTION_HEADER_MOBILE_LAYOUTS.VERTICAL,
+			validator: (value) => Object.values(SECTION_HEADER_MOBILE_LAYOUTS).includes(value),
+		},
 	},
+	emits: ['info-click', 'update:isExpanded'],
 	data() {
 		return {
-			isExpanded: true,
+			isSectionHeaderExpanded: this.isExpanded,
```

Comment:
Let's leave a comment to remove this additional variable when upgrading do new Storybook version

Reply:
Thanks for reminding! I removed it here as it's not really needed. The comment should be added in the Switch component: https://github.com/bethinkpl/design-system/pull/347